### PR TITLE
Bugfixing/track

### LINKED ISSUE #17: FE: When User returns to form page or form page refreshes, User should see submitted mood and habits

### PR DESCRIPTION
**Pull request by woaishenzhen**
### Features
- Bug hunting for dashboard & track page
- Start track testing
- Add some styles to weekly cards

### Refactors 
- 

### Issues and Technical Debt
- Need BE to fix empty daily mood query result

### Other Comments, Concerns, or Questions
- closes #17 

### Checklist
- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature work
- [x] New and existing unit tests pass locally with my changes
- [ ] I ran Lighthouse and WAVE accessibility audits and fixed errors/warnings
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by TannerDale — APPROVED** [review]
Look at those tests!

**Review by WMudgeEllis — APPROVED** [review]

File: src/components/HabitForm/HabitForm.tsx [hunk 1]
```diff
@@ -46,20 +45,23 @@ const HabitForm = () => {
...
-        )}
-        <form className="habit-form" onSubmit={createHabitEntries}>
-          {data && displayHabits()}
-          <button className="habit-submit-button" type="submit">
-            Submit
-          </button>
-        </form>
-      </section>
+      {error ? (
+        <h2>{`Error! ${error.message}`}</h2>
```
Comment (nzambonivergara) [inline src/components/HabitForm/HabitForm.tsx:49]: What does this error.message say? Should we write our own error message for the user?

File: src/utils/context.tsx [hunk 2]
```diff
@@ -83,6 +89,17 @@ const ContextProvider = ({ children }: React.PropsWithChildren<{}>) => {
     return yyyy + "-" + mm + "-" + dd
   }
 
+  const { loading, error, data } = useQuery(QUERY_DAILY_ENTRIES)
+
+  useEffect(() => {
```
Comment (nzambonivergara) [inline src/utils/context.tsx:94]: How often does this run? I'm assuming having it in our Context is what allows our app state to be in sync with the data. Nice solution!

**Review by nzambonivergara — APPROVED** [review]
Thanks for fixing this bug! 🙌🏽

File: src/components/HabitForm/HabitForm.tsx (hunk 1 above)
Comment (woaishenzhen) [reply src/components/HabitForm/HabitForm.tsx:49]: I only saw one "failed to fetch" so far. Not sure if backend has other error message for different status codes

**Pull request merged**